cg/apps/demultiplex/sample_sheet/models.py

```diff
@@ -66,6 +66,15 @@ class FlowCellSampleBCLConvert(FlowCellSample):
 class SampleSheet(BaseModel):
     samples: List[FlowCellSample]
 
+    def get_non_pooled_lanes_and_samples(self) -> Iterable[Tuple[int, str]]:
+        """Return tuples of non pooled lane and sample ids."""
+        non_pooled_lane_sample_id_pairs: List[Tuple[int, str]] = []
+        non_pooled_samples: List[FlowCellSample] = self.get_non_pooled_samples()
+        for sample in non_pooled_samples:
+            sample_id: str = sample.sample_id.split("_")[0]
```

Comment:
Maybe this should be added as a Pydantic validator

Reply:
Sure, the `remove_index_from_sample_id` function can be reused here. I think making this a validator is out of scope for this PR, it required updating more models than I expected